plugins/modules/aci_cloud_aws_provider.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, nkatarmal-crest <nirav.katarmal@crestdatasys.com>
+# Copyright: (c) 2021, Shreyas Srish <ssrish@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: aci_cloud_aws_provider
+short_description: Manage Cloud AWS Provider (cloud:AwsProvider)
+description:
+- Manage AWS provider on Cisco Cloud ACI.
+author:
+- Devarshi Shah (@devarshishah3)
+options:
+  access_key_id:
+    description:
+    - Cloud Access Key ID.
+    type: str
+  account_id:
+    description:
+    - AWS Account ID.
+    type: str
+  email:
+    description:
+    - Account Email Address.
+    type: str
+  http_proxy:
+    description:
+    - Http Proxy to connect to cloud provider.
+    type: str
+  is_account_in_org:
+    description:
+    - Is Account in Organization.
+    type: bool
+  is_trusted:
+    description:
+    - Trusted Tenant
+    type: bool
+  provider_id:
+    description:
+    - AWS provider id
+    type: str
+  region:
+    description:
+    - AWS Region.
+    type: str
+  secret_access_key:
+    description:
+    - Cloud Secret Access Key.
+    type: str
+  tenant:
+    description:
+    - Name of tenant.
+    type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment:
+  - cisco.aci.aci
+
+notes:
+  - More information about the internal APIC class B(cloud:AwsProvider) from
+  - L(the APIC Management Information Model reference,https://developer.cisco.com/docs/apic-mim-ref/).
+'''
+
+EXAMPLES = r'''
+- name: Create aws provider again after deletion as not trusted
+  cisco.aci.aci_cloud_aws_provider:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: ansible_test
+    account_id: 111111111111
+    is_trusted: yes
+    state: present
+  delegate_to: localhost
+
+- name: Delete aws provider
+  cisco.aci.aci_cloud_aws_provider:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: ansible_test
+    account_id: 111111111111
+    is_trusted: yes
+    state: absent
+  delegate_to: localhost
+
+- name: Query aws provider
+  cisco.aci.aci_cloud_aws_provider:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: ansible_test
+    state: query
+  delegate_to: localhost
```

Comment:
Add a query all example (without the tenant)

Reply:
That would indicate all the tenants associated with AWS are displayed which is not the case. Is that desirable?